### PR TITLE
Remove outdated Java code

### PR DESCRIPTION
**Pull request by ipetr0v**
This PR:
- Removes outdated Java code corresponding to Interactive Attestation
- Moves Android and Weather clients from `oak_functions` to `client`

Ref https://github.com/project-oak/oak/issues/3644

### CODE REVIEW

File: java/src/main/java/com/google/oak/transparent_release/RekorLogEntry.java [hunk 1]
```diff
@@ -14,7 +14,7 @@
 // limitations under the License.
 //
 
-package com.google.oak.rekor;
+package com.google.oak.transparent_release;
```
Comment (rbehjati) [inline java/src/main/java/com/google/oak/transparent_release/RekorLogEntry.java:17]: Can we come up with a better package name? How about `transparency`?

**Review by rbehjati — APPROVED** [review]

File: java/src/main/java/com/google/oak/transparent_release/RekorLogEntry.java (hunk 1 above)
Comment (ipetr0v) [reply java/src/main/java/com/google/oak/transparent_release/RekorLogEntry.java:17]: Done

File: java/src/test/java/com/google/oak/client/weather_lookup_client/Main.java
Comment (tiziano88) [inline java/src/test/java/com/google/oak/client/weather_lookup_client/Main.java:1]: I thought we agreed that binaries (even if only used for tests) should be under java/main, and java/test should contain only actual tests?

Comment (ipetr0v) [reply java/src/test/java/com/google/oak/client/weather_lookup_client/Main.java:1]: Moved to `main` `client`

**Pull request merged**